components/datadog/agent/kubernetes.go

```diff
@@ -0,0 +1,55 @@
+package agent
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/datadog/kubernetesagentparams"
+)
+
+type KubernetesAgentOutput struct {
+	components.JSONImporter
+
+	AgentInstallNameLinux   string `json:"agentInstallNameLinux"`
+	AgentInstallNameWindows string `json:"agentInstallNameWindows"`
+}
+
+// DockerAgent is a Docker installer on a remote Host
+type KubernetesAgent struct {
+	pulumi.ResourceState
+	components.Component
+
+	AgentInstallNameLinux   pulumi.StringOutput `pulumi:"agentInstallNameLinux"`
+	AgentInstallNameWindows pulumi.StringOutput `pulumi:"agentInstallNameWindows"`
+}
+
+func (h *KubernetesAgent) Export(ctx *pulumi.Context, out *KubernetesAgentOutput) error {
+	return components.Export(ctx, h, out)
+}
+
+func NewKubernetesAgent(e config.CommonEnvironment, clusterName string, kubeProvider *kubernetes.Provider, options ...kubernetesagentparams.Option) (*KubernetesAgent, error) {
```

Comment:
`clusterName` can be misleading as the Agent `cluster_name / DD_CLUSTER_NAME` parameter while it's only the name of the `KubernetesAgent` resource.

Reply:
Changed to a less misleading parameter name.
https://github.com/DataDog/test-infra-definitions/pull/618/commits/4f275a5004a5dd241119c1fd9765ec095ccfe92a